platform/view/core/manager/manager.go

```diff
@@ -183,25 +183,22 @@ func (cm *manager) Initiate(id string) (interface{}, error) {
 		return nil, errors.Errorf("initiator not found for [%s]", id)
 	}
 
-	return cm.InitiateViewWithIdentity(res.View, cm.me())
+	return cm.InitiateViewWithIdentity(res.View, cm.me(), ctx)
 }
 
-func (cm *manager) InitiateView(view view.View) (interface{}, error) {
-	return cm.InitiateViewWithIdentity(view, cm.me())
+func (cm *manager) InitiateView(view view.View, ctx context.Context) (interface{}, error) {
+	return cm.InitiateViewWithIdentity(view, cm.me(), ctx)
 }
 
-func (cm *manager) InitiateViewWithIdentity(view view.View, id view.Identity) (interface{}, error) {
-	// Create the context
-	cm.contextsSync.Lock()
-	ctx := cm.ctx
-	cm.contextsSync.Unlock()
-	if ctx == nil {
-		ctx = context.Background()
-	}
-	viewContext, err := NewContextForInitiator("", ctx, cm.sp, cm.commLayer, cm.endpointService, id, view, cm.viewTracer)
+func (cm *manager) InitiateViewWithIdentity(view view.View, id view.Identity, ctx context.Context) (interface{}, error) {
+	newCtx, span := cm.viewTracer.Start(ctx, "initiate_view_with_identity")
```

Comment:
could we start this trace by adding to the name also the ID of the viewContext?

Reply:
We can add it as an attribute. Let's take this up in a separate PR